spring-tx/src/main/java/org/springframework/transaction/reactive/TransactionalOperatorImpl.java

```diff
@@ -70,40 +71,19 @@ public ReactiveTransactionManager getTransactionManager() {
 		return this.transactionManager;
 	}
 
-	@Override
-	public <T> Mono<T> transactional(Mono<T> mono) {
-		return TransactionContextManager.currentContext().flatMap(context -> {
-			Mono<ReactiveTransaction> status = this.transactionManager.getReactiveTransaction(this.transactionDefinition);
-			// This is an around advice: Invoke the next interceptor in the chain.
-			// This will normally result in a target object being invoked.
-			// Need re-wrapping of ReactiveTransaction until we get hold of the exception
-			// through usingWhen.
-			return status.flatMap(it -> Mono.usingWhen(Mono.just(it), ignore -> mono,
-					this.transactionManager::commit, (res, err) -> Mono.empty(), this.transactionManager::rollback)
-					.onErrorResume(ex -> rollbackOnException(it, ex).then(Mono.error(ex))));
-		})
-		.contextWrite(TransactionContextManager.getOrCreateContext())
-		.contextWrite(TransactionContextManager.getOrCreateContextHolder());
-	}
-
 	@Override
 	public <T> Flux<T> execute(TransactionCallback<T> action) throws TransactionException {
-		return TransactionContextManager.currentContext().flatMapMany(context -> {
-			Mono<ReactiveTransaction> status = this.transactionManager.getReactiveTransaction(this.transactionDefinition);
-			// This is an around advice: Invoke the next interceptor in the chain.
-			// This will normally result in a target object being invoked.
-			// Need re-wrapping of ReactiveTransaction until we get hold of the exception
-			// through usingWhen.
-			return status.flatMapMany(it -> Flux
-					.usingWhen(
-							Mono.just(it),
```

Comment:
I would also have a preference for this option because the `RuntimeException` feels a bit synthetic, it's only there due to the `usingWhen` implementation, and it doesn't align with the behavior of non-reactive transactions.

I have applied the change, it should be ready for review :)